pkg/logentry/stages/labels.go

```diff
@@ -68,7 +68,7 @@ func (l *labelStage) Process(labels model.LabelSet, extracted map[string]interfa
 			s, err := getString(lValue)
 			if err != nil {
 				if Debug {
-					level.Debug(l.logger).Log("msg", "failed to convert extracted label value to string", "err", err, "type", reflect.TypeOf(lValue).String())
+					level.Debug(l.logger).Log("msg", "failed to convert extracted label value to string", "err", err, "type", reflect.TypeOf(lValue))
```

Comment:
Not sure if we require `.String()` here. If required, we will have to handle `nil` case.

Reply:
This also happens in `template` and `tenant` stage. 
May be we should just set `extracted[n] = ""` here https://github.com/grafana/loki/blob/master/pkg/logentry/stages/json.go#L153

Reply:
Can you paste the stacktrace here to verify this was the problem ?

Reply:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xf8 pc=0x1978f03]

goroutine 165 [running]:
testing.tRunner.func1(0xc0001b8700)
	/usr/local/Cellar/go/1.13.3/libexec/src/testing/testing.go:874 +0x3a3
panic(0x1aa2160, 0x248be70)
	/usr/local/Cellar/go/1.13.3/libexec/src/runtime/panic.go:679 +0x1b2
github.com/grafana/loki/pkg/logentry/stages.(*labelStage).Process(0xc00028ffe0, 0xc0002bd5f0, 0xc0002bd650, 0xc000302040, 0xc0002e0940)
	/Users/adityacs/adityacs/loki/src/github.com/grafana/loki/pkg/logentry/stages/labels.go:71 +0x223
github.com/grafana/loki/pkg/logentry/stages.(*Pipeline).Process(0xc0002b7cc0, 0xc0002bd5f0, 0xc0002bd650, 0xc000302040, 0xc0002e0940)
	/Users/adityacs/adityacs/loki/src/github.com/grafana/loki/pkg/logentry/stages/pipeline.go:93 +0x213
github.com/grafana/loki/pkg/logentry/stages.TestLabelsPipeline_Labels(0xc0001b8700)
	/Users/adityacs/adityacs/loki/src/github.com/grafana/loki/pkg/logentry/stages/labels_test.go:47 +0x271
testing.tRunner(0xc0001b8700, 0x1c3a568)
	/usr/local/Cellar/go/1.13.3/libexec/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/Cellar/go/1.13.3/libexec/src/testing/testing.go:960 +0x350

```

Reply:
This line points out the error `github.com/grafana/loki/pkg/logentry/stages.(*labelStage).Process(0xc00028ffe0, 0xc0002bd5f0, 0xc0002bd650, 0xc000302040, 0xc0002e0940)
/Users/adityacs/adityacs/loki/src/github.com/grafana/loki/pkg/logentry/stages/labels.go:71 +0x223`

Reply:
I see you should add a regression test then.